fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java

```diff
@@ -36,8 +36,13 @@ public abstract class MixinTurtleMoveCommand {
                     .stream(ValkyrienSkies.positionToNearbyShipsAndWorld(world, position.getX() + 0.5, position.getY() + 0.5, position.getZ() + 0.5, 0.1))
                     .map(pos -> ValkyrienSkies.getShipManagingBlock(world, pos))
                     .map(s -> ValkyrienSkies.positionToShip(s, new Vector3d(position.getX() + 0.5, position.getY() + 0.5, position.getZ() + 0.5)))
-                    .map(pos -> world.getBlockState(new BlockPos(ValkyrienSkies.toMinecraft(pos))))
-                    .anyMatch(BlockState::isAir);
+                    .map(pos -> new BlockPos(ValkyrienSkies.toMinecraft(pos)))
+                    // Ignore turtle itself
+                    .filter(pos -> !pos.equals(turtlePlayer.blockPosition()))
```

Comment:
I do not know if there are better way to do this, because blockPosition seems will return the turtle position in actual world, but it should return a shipyard position I'm not sure?